packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -175,10 +216,31 @@ export function Pagination<NodesType>({
   });
 }
 
-function getParamsWithoutPagination(paramsString?: string) {
+function getParamsWithoutPagination(
+  paramsString?: string,
+  state?: PaginationState<any>,
+) {
   const params = new URLSearchParams(paramsString);
-  params.delete('cursor');
-  params.delete('direction');
+
+  // Get all namespaces from state
+  const activeNamespaces = Object.keys(state?.pagination || {});
+
+  // Handle non-namespaced params (empty string namespace)
+  if (activeNamespaces.includes('')) {
+    params.delete('cursor');
+    params.delete('direction');
+  }
+
+  activeNamespaces
+    .filter((namespace) => namespace !== '')
+    .forEach((namespace) => {
+      const cursorParam = `${namespace}_cursor`;
+      const directionParam = `${namespace}_direction`;
+
+      params.delete(cursorParam);
+      params.delete(directionParam);
+    });
+
```

Comment:
Optionally, you could roll this into one by doing:
```
      const namespacePrefix = namespace === '' ? '' : `${namespace}_`
      const cursorParam = `${namespacePrefix}cursor`;
      const directionParam = `${namespacePrefix}direction`;
```
Though tbh it would have been fine to give no special treatment to the empty namespace, and just used `_cursor` and `_direction` when working with it (although granted that might look ugly in the URL)

Reply:
And yeah I thought `_cursor` was a bit of an eye sore + would break existing links when folks upgrade.